Integrations/integration-VirusTotal.yml

```diff
@@ -0,0 +1,519 @@
+        }
+        if (longFormat === 'true') {
+            for (var j=0; j<arrTitle.length; j++) {
+                if (arrTitle[j].a) {
+                    md += '### ' + arrTitle[j].t + '\n';
+                    // Print only the first 10 rows
+                    var curr = [];
+                    for (var k=0; k<Math.min(arrTitle[j].a.length, sampleSize); k++) {
+                        curr.push(arrTitle[j].a[k]);
+                    }
+                    md += arrToMd(curr) + '\n';
+                }
+            }
+        }
+        return {Type: entryTypes.note, Contents: res.body, ContentsFormat: formats.json, HumanReadable: md, EntryContext: ec};
+    };
+
+    var doURL = function(url, threshold, longFormat, sampleSize, submitWait) {
+        if (!submitWait) {
+            submitWait = 0;
+        }
+        if (!sampleSize) {
+            sampleSize = 10;
+        }
+        if (!threshold) {
+            threshold = 10;
+        }
+        var res = doReq('POST', 'url/report', {resource: url, scan:1});
+
+        var o = res.obj;
+        var ec = {};
+        var md = '## VirusTotal URL Reputation for: ' + url + '\n';
+
+        if (!o.scans && submitWait>0) {
+            wait(parseInt(submitWait));
+            res = doReq('GET', 'url/report', {resource: url});
+            o = res.obj;
+        }
+        if (!o.scans) {
+            md += 'URL submitted for scan. Please retry command later\n';
+            md += 'Scan ID: **' + o.scan_id + '**\n\n';
+        } else {
+            md += 'Last scan date: *' + o.scan_date + '*\n';
+            md += 'Scan ID: **' + o.scan_id + '**\n\n';
+            md += 'Total scans: **' + o.total + '**\n';
+            md += 'Positive scans: **' + o.positives + '**\n';
+
+            if (longFormat === 'true') {
+                md += '### Scans\n';
+                // Print only the first 10 rows
+                var curr = [];
+                for (var i=0; i<Math.min(Object.keys(o.scans).length, sampleSize); i++) {
+                    curr.push({Site: Object.keys(o.scans)[i], Detected: o.scans[Object.keys(o.scans)[i]].detected, Result: o.scans[Object.keys(o.scans)[i]].result});
+                }
+                md += arrToMd(curr) + '\n';
+            }
+            if (o.positives >= threshold) {
+                if (!ec.badUrls) {
+                    ec.badUrls = [];
+                }
+                ec.badUrls.push(o.url);
+            }
+        }
+        return {Type: entryTypes.note, Contents: res.body, ContentsFormat: formats.json, HumanReadable: md, EntryContext: ec};
+    };
+
+    var doDomain = function(domain, longFormat, sampleSize) {
+        if (!sampleSize) {
+            sampleSize = 10;
+        }
+        var res = doReq('GET', 'domain/report', {domain: domain});
+        var o = res.obj;
+        var ec = {};
+        var md = '## VirusTotal Domain Reputation for: ' + domain + '\n';
+        md += '#### Domain categories: *' + o.categories + "*\n";
+
```

Comment:
I think it's better to separate the call and the parsing... in this file they are all coupled together.